action.yml

```diff
@@ -26,14 +26,18 @@ inputs:
     description: 'Check if a cache entry exists for the given input(s) (key, restore-keys) without downloading the cache'
     default: 'false'
     required: false
+  save-always:
+    description: 'Run the post step to save the cache even if another step before fails'
+    default: 'false'
+    required: false    
 outputs:
   cache-hit:
     description: 'A boolean value to indicate an exact match was found for the primary key'
 runs:
   using: 'node16'
   main: 'dist/restore/index.js'
   post: 'dist/save/index.js'
-  post-if: success()
+  post-if: "success() || github.event.inputs.save-always"
```

Comment:
```suggestion
  post-if: "github.event.inputs.save-always"
```

Reply:
I'm sorry I should give more context. 
I think we need to revert this change. I was thinking we can pass `success()` as an arg but it will not work. I found a workaround for my case and manually run save restore action with different conditions.

Reply:
Reverted with https://github.com/actions/cache/pull/1242/commits/0e9aed0d0d8123bb9d12426b86a761cf9d0c4cff